src/mlnet/CodeGenerator/CodeGenerationHelper.cs

```diff
@@ -187,6 +189,8 @@ public void GenerateCode()
                             logger.Log(LogLevel.Error, Strings.UnsupportedMlTask);
                             break;
                     }
+
+                    t.IsBackground = true;
```

Comment:
While we have control that our threads are background threads, we don't know that (of have control of) the learner's threads are background threads.

This will lead to not exiting properly.

Reply:
Yeah that could be a problem.

Reply:
I would think that any threads spawned by ML.NET should be background threads. And if they aren't, I would venture a guess that it is a bug that should be fixed in the learners.

Reply:
Yep looks like all of them are background threads except for: 
https://github.com/dotnet/machinelearning/blob/6ebfeb33cd891db7328d032c903c3571d91a9265/src/Microsoft.ML.FastTree/SumupPerformanceCommand.cs#L291

Looks like this is the only place where a foreground thread is started. This one looks like a command used by some tool to measuring performance. So that should be okay I guess.

Reply:
@TomFinley @codemzs - any thoughts on why this thread needs to run on the foreground?

Reply:
Oh, this is an `ICommand`, which will only run from the MAML program. I don't think it is possible to run this code from a public API in ML.NET.

Reply:
Looking even further, this appears to be more of a baseline test than anything.  Maybe `SumupPerformanceCommand` can just be deleted and/or turned into a BenchmarkDotNet test instead. I don't see why this code needs to be part of the shipping product.

Reply:
Any idea what occurs with LightGBM? Is there other outside could we are unsure of?

Seems simple enough to stay with `Environment.Exit(exitcode)` and be more future proof.

Reply:
Environment.Exit isn't the "typical" way of terminating the process.

What do you expect would happen with this code:

```C#
IDisposable resource = ....;
try
{
   // use the resource
   // ...

   Environment.Exit(0);
}
finally
{
   resource.Dispose();
}
```

or even:

```C#
using (IDisposable resource = ....)
{
   // use the resource
   // ...

   Environment.Exit(0);
}
```

Would you expect the resource to be disposed?

Reply:
Isn't disposing only meaningful if the process is going to stay alive?  The resources belonging to exiting process will automatically released by OS ?

Reply:
It depends on what you are disposing. `Dispose()` could be closing a connection somewhere else, or flushing log files so they get written to disk.